src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp

```diff
@@ -33,40 +33,46 @@ medDatabaseSettingsWidget::medDatabaseSettingsWidget(QWidget *parent) :
     setTabName("Database");
 
     d->dbPath = new QLineEdit(this);
-    d->btChooseDir = new QPushButton(tr("Select directory..."), this);
+    d->selectDirectoryButton = new QPushButton(tr("Select directory..."), this);
+    d->selectDirectoryButton->setToolTip("Change the database directory or create a new database by selecting an empty directory");
 
     QWidget* databaseLocation = new QWidget(this);
     QHBoxLayout* dbLayout = new QHBoxLayout;
     dbLayout->addWidget(d->dbPath);
-    dbLayout->addWidget(d->btChooseDir);
+    dbLayout->addWidget(d->selectDirectoryButton);
     databaseLocation->setLayout(dbLayout);
 
-    connect(d->btChooseDir, SIGNAL(clicked()), this, SLOT(selectDbDirectory()));
+    connect(d->selectDirectoryButton, SIGNAL(clicked()), this, SLOT(selectDirectory()));
 
     databaseLocation->setContentsMargins(0,-8,0,0);
 
     QFormLayout* formLayout = new QFormLayout(this);
     formLayout->addRow(tr("Database location:"), databaseLocation);
+    formLayout->addRow(new QLabel("* The changes will only be effective after saving and restarting the application."));
 }
 
-void medDatabaseSettingsWidget::selectDbDirectory()
+void medDatabaseSettingsWidget::selectDirectory()
 {
-    QFileDialog dialog(this);
-    dialog.setFileMode(QFileDialog::DirectoryOnly);
-
-     QString path;
-     if (dialog.exec())
-     {
-         QString path = dialog.selectedFiles().first();
-         if (!QDir(path).entryInfoList(QDir::NoDotAndDotDot | QDir::Dirs | QDir::Files).isEmpty())
-         {
-             QMessageBox::information( this, tr("Directory not empty"), tr("The archive directory needs to be empty! \nPlease choose another one."));
-         }
-         else
-         {
-             d->dbPath->setText(path);
-         }
-     }
+    QDir defaultPath = QDir(medStorage::dataLocation());
+    defaultPath.cdUp();
+
+    QString path = QFileDialog::getExistingDirectory(this, tr("Select the database directory"),
+                                                     QDir(defaultPath).absolutePath(),
+                                                     QFileDialog::ShowDirsOnly
+                                                     | QFileDialog::DontResolveSymlinks);
+
+    if(path != "") //the user hasn't clicked on Cancel
+    {
+        if (!QDir(path).entryInfoList(QDir::NoDotAndDotDot | QDir::Dirs | QDir::Files).isEmpty() && QDir(path).entryInfoList(QStringList("db")).isEmpty())
+        {
+            QMessageBox::information( this, tr("Database location not valid"),
+                                      tr("The directory selected is not valid:\nEither it doesn't point to an existing medInria database directory \nOR your new directory is not empty. \nPlease choose another one."));
```

Comment:
Or maybe: "The directory selected is not valid:\nEither it doesn't point to an existing database directory \nOR your new directory is not empty. \nPlease choose another one."
No need of the software name :)